DVConstraints.py

```diff
@@ -2553,6 +2713,196 @@ def writeTecplot(self, handle):
         for i in range(len(self.coords)//2):
             handle.write('%d %d\n'% (2*i+1, 2*i+2))
 
+class RadiusConstraint(GeometricConstraint):
+    """
+    DVConstraints representation of a set of radius of curvature
+    constraints. One of these objects is created each time a
+    addLERadiusConstraints call is made. The user should not have
+    to deal with this class directly.
+    """
+
+    def __init__(self, name, coords, lower, upper, scaled, scale, DVGeo,
+                 addToPyOpt):
+        self.name = name
+        self.coords = coords
+        self.nCon = len(self.coords)//3
+        self.lower = lower
+        self.upper = upper
+        self.scaled = scaled
+        self.scale = scale
+        self.DVGeo = DVGeo
+        self.addToPyOpt = addToPyOpt
+
+        GeometricConstraint.__init__(self, self.name, self.nCon, self.lower,
+                                     self.upper, self.scale, self.DVGeo,
+                                     self.addToPyOpt)
+
+        # First thing we can do is embed the coordinates into DVGeo
+        # with the name provided:
+        self.DVGeo.addPointSet(self.coords, self.name)
+
+        # Now get the reference lengths
+        self.r0, self.c0 = self.computeCircle(self.coords)
+
+    def splitPointSets(self, coords):
+        p1 = coords[:self.nCon]
+        p2 = coords[self.nCon:self.nCon*2]
+        p3 = coords[self.nCon*2:]
+        return p1, p2, p3
+
+    def computeReferenceFrames(self, coords):
+        p1, p2, p3 = self.splitPointSets(coords)
+
+        # Compute origin and unit vectors (xi, eta) of 2d space
+        origin = (p1 + p2) / 2.0
+        nxi = p1 - origin
+        neta = p3 - origin
+        for i in range(self.nCon):
+            nxi[i] /= geo_utils.euclideanNorm(nxi[i])
+            neta[i] /= geo_utils.euclideanNorm(neta[i])
+
+        # Compute component of eta in the xi direction
+        eta_on_xi = numpy.einsum('ij,ij->i', nxi, neta)
+        xi_of_eta = numpy.einsum('ij,i->ij', nxi, eta_on_xi)
+
+        # Remove component of eta in the xi direction
+        neta = neta - xi_of_eta
+        for i in range(self.nCon):
+            neta[i] /= geo_utils.euclideanNorm(neta[i])
+
+        return origin, nxi, neta
+
+    def computeCircle(self, coords):
```

Comment:
Can you include some short text that describes the mathematical formulation here? A reference also works.